test/tests/carousel_carousel-2-tablist.js

```diff
@@ -0,0 +1,364 @@
+  const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+  await tabs[index].click();
+};
+
+const waitAndCheckFocus = async function (t, selector, index) {
+  return t.context.session.wait(async function () {
+    return t.context.session.executeScript(function () {
+      const [selector, index] = arguments;
+      let items = document.querySelectorAll(selector);
+      return items[index] === document.activeElement;
+    }, selector, index);
+  }, t.context.waitTime, 'Timeout waiting for document.activeElement to become item at index ' + index + ' of elements selected by: ' + selector);
+};
+
+const waitAndCheckAriaSelected = async function (t, index) {
+  return t.context.session.wait(async function () {
+    const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+    return (await tabs[index].getAttribute('aria-selected')) === 'true';
+  }, t.context.waitTime, 'Timeout waiting for aria-selected to be set to true.');
+};
+
+// Attributes
+
+ariaTest('Carousel 2: rotation button has aria-label that is updated based on pause state', exampleFile, 'rotation-aria-label', async (t) => {
+  t.plan(4);
+  await assertAriaLabelExists(t, ex.rotationSelector);
+  await assertAttributeValues(t, ex.rotationSelector, 'aria-label', ex.rotationLabelPlaying);
+
+  let rotationButtonEl = await t.context.session.findElement(By.css(ex.rotationSelector));
+
+  // Send SPACE key and wait for change of 'aria-label'
+  await rotationButtonEl.sendKeys(Key.SPACE);
+  await t.context.session.wait(async function () {
+    return rotationButtonEl.getAttribute('aria-label') !== ex.rotationLabelPlaying;
+  }, t.context.waitTime, 'Timeout waiting for rotation button\'s aria-label to change');
+
+  await assertAttributeValues(t, ex.rotationSelector, 'aria-label', ex.rotationLabelPaused);
+
+  // Send ENTER key and wait for change of 'aria-label'
+  await rotationButtonEl.sendKeys(Key.ENTER);
+  await t.context.session.wait(async function () {
+    return rotationButtonEl.getAttribute('aria-label') !== ex.rotationLabelPaused;
+  }, t.context.waitTime, 'Timeout waiting for rotation button\'s aria-label to change');
+
+  await assertAttributeValues(t, ex.rotationSelector, 'aria-label', ex.rotationLabelPlaying);
+
+
+});
+
+ariaTest('Carousel 2: role="tablist" on div element', exampleFile, 'tablist-role', async (t) => {
+  t.plan(1);
+  await assertAriaRoles(t, 'ex1', 'tablist', '1', 'div');
+});
+
+ariaTest('Carousel 2: "aria-label" attribute on role="tablist"', exampleFile, 'tablist-aria-label', async (t) => {
+  t.plan(1);
+  await assertAriaLabelExists(t, ex.tablistSelector);
+});
+
+ariaTest('Carousel 2: role="tab" on button elements', exampleFile, 'tab-role', async (t) => {
+  t.plan(1);
+  await assertAriaRoles(t, 'ex1', 'tab', ex.tabCount, 'button');
+});
+
+ariaTest('"Carousel 2: aria-selected" set on role="tab"', exampleFile, 'tab-aria-selected', async (t) => {
+  t.plan(2 * ex.tabCount * ex.tabCount);
```

Comment:
Since the logic being tested here isn't unique to tab index, would it make sense to test before and after only one click, and then perhaps after changing with the keyboard as well?

Reply:
Thanks @spectranaut, that makes sense to me!

How do you feel about the other suggestion, to only tests before/after one or two clicks instead of all of them? Testing every tab seems unnecessary, since there's nothing in the code that would run differently for different tabs.